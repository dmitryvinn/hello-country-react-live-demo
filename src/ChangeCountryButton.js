import React from "react";

const buttonStyle = {
    border: 'none',
    color: '#ffffff',
    padding: '15px 32px',
    textAlign: 'center',
    textDecoration: 'none',
    display: 'inlineBlock',
    fontSize: '17px',
    backgroundColor: '#555555'
};

class ChangeCountryButton extends React.Component {
    render() {
        return <button className="changeCountry"
                       style={buttonStyle}
                       onClick={this.props.action}>Switch  Country</button>
    }
}

export default ChangeCountryButton;
