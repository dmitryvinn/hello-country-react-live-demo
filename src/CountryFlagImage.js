import React from "react";
var MediaQuery = require("react-responsive").default;

const imageStyle = {
    height: "300px",
    width: "550px",
    border: "2px solid"
};

class CountryFlagImage extends React.Component {
    render() {
        return (
            <div>
                <MediaQuery query="screen and (min-width: 376px)">
                    <img
                        className="countryFlag"
                        style={imageStyle}
                        src={this.props.img}
                        alt="Country Flag"
                    />
                </MediaQuery>

                <MediaQuery query="screen and (width: 375px)">
                    <img
                        className="countryFlag"
                        style={{ visibility: "hidden" }}
                        src={this.props.img}
                        alt="Country Flag"
                    />
                </MediaQuery>
                <MediaQuery query="screen and (max-width: 374px)">
                    <img
                        className="countryFlag"
                        style={{ display: "none" }}
                        src={this.props.img}
                        alt="Country Flag"
                    />
                </MediaQuery>
            </div>
        );
    }
}

export default CountryFlagImage;