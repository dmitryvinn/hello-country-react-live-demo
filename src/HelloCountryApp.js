import React from "react";
import CountryFlagImage from "./CountryFlagImage";
import WelcomeCountryText from "./WelcomeCountryText";
import ChangeCountryButton from "./ChangeCountryButton";

class HelloCountryApp extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            country: "american",
            label: "America",
            flag: "https://i.imgur.com/c56iDt7.png"
        };
        this.ChangeCountryButton = this.ChangeCountryButton.bind(this);
    }

    ChangeCountryButton() {
        var countryName = this.state.country;
        var CountryFlagImage = this.state.flag;
        var countryLabel = this.label;
        if (countryName === "america") {
            countryName = "america";
            countryLabel = "America";
            CountryFlagImage = "https://i.imgur.com/c56iDt7.png";
        } else {
            countryName = "canada";
            countryLabel = "Canada";
            CountryFlagImage = "https://i.imgur.com/bh3BkeN.png";
        }
        this.setState({
            country: countryName,
            flag: CountryFlagImage,
            label: countryLabel
        });
    }

    render() {
        return (
            <div>
                <WelcomeCountryText name={this.state.label} />
                <CountryFlagImage img={this.state.flag} />
                <div>
                    <ChangeCountryButton action={this.ChangeCountryButton} />
                </div>
            </div>
        );
    }
}

export default HelloCountryApp;
