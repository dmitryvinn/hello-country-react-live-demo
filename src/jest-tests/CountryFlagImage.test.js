import React from 'react';
import renderer from 'react-test-renderer';
import ChangeCountryButton from "../ChangeCountryButton";
import HelloCountryApp from "../HelloCountryApp";
import CountryFlagImage from "../CountryFlagImage";

test('Validate Country Flag', () => {
    // Initialize Component
    let component;
    component = renderer.create
    (
        <CountryFlagImage/>
    );
    // Format result
    let tree = component.toJSON();
    // Assert
    expect(tree).toMatchSnapshot();
});