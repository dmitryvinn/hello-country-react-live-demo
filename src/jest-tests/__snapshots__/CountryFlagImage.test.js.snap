// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country Change Button  1`] = `<div />`;

exports[`Validate Country Flag 1`] = `<div />`;
